packages/component-library-angular/src/checkbox/Checkbox.spec.ts

```diff
@@ -0,0 +1,365 @@
+import { By } from '@angular/platform-browser';
+import { fireEvent, render, screen } from '@testing-library/angular';
+import { clearElements } from '../utils';
+import { UtrechtCheckboxAttr } from './component';
+import '@testing-library/jest-dom';
+
+afterEach(() => {
+  clearElements();
+});
+
+describe('Checkbox', () => {
+  it('renders a checkbox role element', async () => {
+    await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    const checkbox = screen.getByRole('checkbox');
+
+    expect(checkbox).toBeInTheDocument();
+    expect(checkbox).toBeVisible();
+  });
+
+  it('renders an HTML input type=checkbox element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    expect(fixture.nativeElement).toBeInTheDocument();
+  });
+
+  it('displays as CSS block element', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+
+    expect(fixture.nativeElement).toBeVisible();
+    expect(fixture.nativeElement).toHaveStyle({ display: 'block' });
+  });
+
+  it('renders a design system BEM block class name', async () => {
+    const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+      declarations: [UtrechtCheckboxAttr],
+    });
+    const debugElement = fixture.debugElement.query(By.css('[type="checkbox"]input'))!;
+    expect(debugElement.nativeElement.classList.contains('utrecht-checkbox')).toBeTruthy();
+  });
+
+  describe('checked variant', () => {
+    it('is not checked by default', async () => {
+      const { fixture } = await render('<input type="checkbox" utrecht-checkbox />', {
+        declarations: [UtrechtCheckboxAttr],
+      });
+
+      expect(fixture.nativeElement).not.toBeChecked();
```

Comment:
For accessibility tree based checks, such as `toBeChecked()`, I would prefer you locate the relevant element using `getByRole('checkbox')`